plugins/modules/files/json_file.py

```diff
@@ -0,0 +1,540 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2022, DEMAREST Maxime <maxime@indelog.fr>
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import json
+import os
+
+from ansible_collections.community.general.plugins.module_utils.common.validation import check_type_dict_or_list
+from ansible_collections.community.general.plugins.module_utils.mh.module_helper_dest_file import (
+    DestFileModuleHelper,
+    ModuleHelperException,
+)
+from ansible_collections.community.general.plugins.module_utils.data_merge_utils import (
+    DataMergeUtils,
+    MergeType,
+    ListDiffType,
+)
+
+
+DOCUMENTATION = r'''
+---
+module: json_file
+short_description: Managage (add, remove, update) elements in a json file.
+description:
+  - This module ensures that a set of values is present or absent in a json
+    file.
+  - The check can be done in all childrens of an object.
+  - It also makes sure a value is present or absent in a list or in a
+    specific index of the list.
+  - In background, this module use the python json lib
+    U(https://docs.python.org/3/library/json.html).
+notes:
+  - Take note of the following warning before using this module, B(if you have
+    some comments in your json file, this module will not preserve them !) So
+    don't use it if it's important for you to keep these comments.
+  - This module supports C(--check) and C(--diff) flags.
+options:
+  path:
+    description:
+      - Absolute path of the json file.
+    type: path
+    required: true
+    alias: [ dest ]
+  state:
+    description:
+      - If set to C(present), that ensure the values are present in the json
+        file.
+      - If set to C(absent), that ensure the values are absent in the json
+        file.
+      - If set to C(identic), that ensure the values in the json are identic to
+        those provided in arguments.
```

Comment:
You should explain what `identic` means in more detail. I would expect that `identic` does the same as `present` and wonder what the differences are.

Also common parameters used by multiple modules should be in a docs fragment instead of being repeated.